core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -272,7 +277,7 @@ public Transaction(NetworkParameters params) {
 
     public Transaction(NetworkParameters params, SpecialTxPayload specialTxPayload) {
         this(params);
-        setVersionAndType(3, specialTxPayload.getType());
+        setVersionAndType(MAX_STANDARD_VERSION, specialTxPayload.getType());
```

Comment:
This should be `DIP2_VERSION`, since `MAX_STANDARD_VERSION` may be raised to 4 at a later time, but this particular constructor is for creating a version 3 transaction which has a Special transaction payload.